docs/src/main/asciidoc/mutiny-primer.adoc

```diff
@@ -0,0 +1,345 @@
+Mutiny is taking a very different angle:
+
+**Event-Driven** -
+Mutiny places events at the core of its design.
+With Mutiny, you observe events, react to them, and create elegant and readable processing pipelines.
+A Ph.D. in functional programming is not required.
+
+**Navigable** - Even with intelligent code completion, classes with hundreds of methods are confusing.
+Mutiny provides a navigable and explicit API driving you towards the operator you need.
+
+**Non-Blocking I/O** - Mutiny is the perfect companion to tame the asynchronous nature of applications with non-blocking I/O (which powers xref:quarkus-reactive-architecture.adoc[Quarkus]).
+Declaratively compose operations, transform data, enforce progress, recover from failures, and more.
+
+**Made for an asynchronous world** - Mutiny can be used in any asynchronous application such as event-driven microservices, message-based applications, network utilities, data stream processing, and of course... reactive applications!
+
+**Reactive Streams and Converters Built-In** - Mutiny is based on the https://www.reactive-streams.org/[Reactive Streams] specification, and so it can be integrated with any other reactive programming library.
+In addition, it proposes converters to interact with other popular libraries.
+
+== Mutiny and the I/O Threads
+
+Quarkus is powered by a xref:quarkus-reactive-architecture.adoc#engine[reactive engine], and when developing a reactive application, your code is executed on one of the few I/O threads.
+Remember, you must never block these threads, and the model would collapse if you do.
+So, you can't use blocking I/O.
+Instead, you need to schedule the I/O operation and pass a continuation.
+
+image::reactive-thread.png[alt=Reactive Execution Model and I/O Threads]
+
+The Mutiny event-driven paradigm is tailored for this.
+When the I/O operation completes successfully, the Uni that represents it emits an item event.
+When it fails, it emits a failure event.
+The continuation is simply and naturally expressed using the event-driven API.
+
+== Mutiny through Examples
+
+Many Quarkus extensions expose Mutiny APIs. In this section, we use the MongoDB extension to illustrate how to use Mutiny.
+
+Let's imagine a simple structure representing an element from the https://en.wikipedia.org/wiki/Periodic_table[periodic table]:
+
+[source, java]
+----
+public class Element {
+
+   public String name;
+   public String symbol;
+   public int position;
+
+   public Element(String name, String symbol, int position) {
+       this.name = name;
+       this.symbol = symbol;
+       this.position = position;
+   }
+
+   public Element() {
+        // Use by JSON mappers
+   }
+}
+----
+
+This structure contains the name, symbol, and position of the element.
+To retrieve and store elements into a Mongo collection, you can use the following code:
+
+[source, java]
+----
+@ApplicationScoped
+public class ElementService {
+
+   @Inject
+   ReactiveMongoClient client;
```

Comment:
Let's use constructor injection as there are still people who think that CDI can't work with it :(

Reply:
Good idea! 
I need to use this feature more (especially when doing live coding)